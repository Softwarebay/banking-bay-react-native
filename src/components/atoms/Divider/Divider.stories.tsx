import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Divider } from './Divider';

storiesOf('Atoms/Divider', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('Default', () => <Divider testID="Divider-test-id" />);
