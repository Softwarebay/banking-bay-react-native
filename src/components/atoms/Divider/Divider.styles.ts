import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	line: {
		borderColor: colors.divider,
		borderTopWidth: 1,
	},
});
