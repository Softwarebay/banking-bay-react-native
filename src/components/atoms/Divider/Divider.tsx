import { StyleProp, View, ViewStyle } from 'react-native';

import { styles } from './Divider.styles';

interface Props {
	style?: StyleProp<ViewStyle>;
	testID?: string;
}

export const Divider = ({ testID, style }: Props) => {
	return <View style={[styles.line, style]} testID={testID} />;
};
