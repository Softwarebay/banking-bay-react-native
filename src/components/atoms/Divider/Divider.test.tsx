import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Divider } from './Divider';

import type { Props } from './Divider';

describe('Divider', () => {
	const props: Props = {
		testID: 'divider-test-id',
	};
	const propsWithStyle: Props = {
		...props,
		style: {
			opacity: 0.8,
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<Divider {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render divider', () => {
		render(<Divider {...props} />);

		const divider = screen.getByTestId('divider-test-id');

		expect(divider).toBeTruthy();
	});

	it('should apply additional styles through props', () => {
		render(<Divider {...propsWithStyle} />);

		const divider = screen.getByTestId('divider-test-id');

		expect(divider).toHaveStyle({ opacity: 0.8 });
	});
});
