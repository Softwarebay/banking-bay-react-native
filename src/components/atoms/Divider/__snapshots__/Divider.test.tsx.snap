// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#CED6E1",
        "borderTopWidth": 1,
      },
      undefined,
    ]
  }
  testID="divider-test-id"
/>
`;
